public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Editor.tsx

```diff
@@ -76,6 +77,18 @@ export const Editor = (props: Props): JSX.Element => {
   const { datasource } = props;
   const { measurement, policy } = query;
 
+  useEffect(() => {
+    let ignore = false;
+    getAllPolicies(datasource).then((result) => {
+      if (!ignore) {
+        setRetentionPolicies(result);
+      }
+    });
+    return () => {
+      ignore = true;
+    };
+  }, [datasource]);
+
```

Comment:
just a random note. i had good experience with using the `react-use` npm package for these situations, for example this function: https://streamich.github.io/react-use/?path=/story/side-effects-useasyncfn--docs may help you to write simpler code here (but maybe not, just an idea)

(`react-use` is already used in grafana so it does not need to be added)

Reply:
I will check it. Thanks for the suggestion.